src/coreComponents/mesh/MeshLevel.cpp

```diff
@@ -121,35 +119,43 @@ MeshLevel::MeshLevel( string const & name,
   m_faceManager->faceNormal() = source.m_faceManager->faceNormal();
 
   // Faces
-//  ArrayOfArraysView< localIndex const > const facesToNodesMapSource = m_faceManager->nodeList().toViewConst();
   ArrayOfArrays< localIndex > & faceToNodeMapNew = m_faceManager->nodeList();
   ArrayOfArraysView< localIndex const > const & facesToEdges = m_faceManager->edgeList().toViewConst();
-  localIndex const estimatedNumNodesPerFace = pow( order+1, 2 );
-  faceToNodeMapNew.resize( faceToNodeMapNew.size(), estimatedNumNodesPerFace );
+  localIndex const numNodesPerFace = pow( order+1, 2 );
 
 
   // add the number of non-edge face nodes
   localIndex numInternalFaceNodes = 0;
+  localIndex const numNonEdgeNodesPerFace = pow( order-1, 2 );
   for( localIndex kf=0; kf<m_faceManager->size(); ++kf )
   {
     localIndex const numEdgesPerFace = facesToEdges.sizeOfArray( kf );
-//    localIndex const numVertexNodesPerFace = facesToNodesMapSource.sizeOfArray( kf );
-//    localIndex const numEdgeNodesPerFace = numVertexNodesPerFace + numEdgesPerFace * numNonVertexNodesPerEdge;
-
     if( numEdgesPerFace==4 )
-    {
-      localIndex const numNonEdgeNodesPerFace = pow( order-1, 2 );
       numInternalFaceNodes += numNonEdgeNodesPerFace;
-
-      localIndex const numNodesPerFace = pow( order+1, 2 );
-      faceToNodeMapNew.resizeArray( kf, numNodesPerFace );
-    }
     else
     {
       GEOSX_ERROR( "need more support for face geometry" );
     }
   }
 
+  // number of elements in each row of the map as capacity
+  array1d< localIndex > counts( faceToNodeMapNew.size());
+  counts.setValues< parallelHostPolicy >( numNodesPerFace );
+
+  //  reconstructs the faceToNodeMap with the provided capacity in counts
+  faceToNodeMapNew.resizeFromCapacities< parallelHostPolicy >( faceToNodeMapNew.size(), counts.data() );
+
+  // setup initial values of the faceToNodeMap using emplaceBackAtomic
+  forAll< parallelHostPolicy >( numNodesPerFace, [faceToNodeMapNew = faceToNodeMapNew.toView()] ( localIndex faceIndex )
+  {
+    for( faceIndex =0; faceIndex < faceToNodeMapNew.size(); faceIndex++ )
+    {
+      faceToNodeMapNew.emplaceBackAtomic< AtomicPolicy< parallelHostPolicy > >( faceIndex, 0 );
+    }
+
+  } );
```

Comment:
Is it possible make this initialization closer to the definition of `faceToNodeMapNew`?
Does it make sense?

Reply:
Hmmm I don't know exactly  what you mean by "closer to the definition" all I know is that initilaize this map as Jie does allows us to divide by 60 the initialization time... (I had the Caliper output inside the comments of the PR)

Reply:
I mean that you declare `faceToNodeMapNew`, and you fill it quite far from this declaration.
Could it be possible to make those two actions close together for readability concerns?
The "_Does it make sense?_" is related to the first sentence, _i.e._ is bringing those two actions closer something which makes sense?

> all I know is that initilaize this map as Jie does allows us to divide by 60 the initialization time... 

Which is awesome.
Did you run the integrated tests so we're sure that it's breaking nothing? (The devil is in the details)
Is this 60 factor something you can find on all cases or is it more related to the specific case you're studying?

Reply:
Yes, I will move it to line 125 to make it adjacent to the declaration of faceToNodeMapNew arrays. And we will run the integrated tests to make sure it will not breaking nothing (devil is the details indeed~~). Thanks for the suggestion!